dj/api/nodes.py

```diff
@@ -329,6 +285,87 @@ def create_node(
     return node  # type: ignore
 
 
+@router.post("/nodes/{name}/{column}/dimension/{dimension}/")
+def add_dimension_to_node(
+    name: str,
+    column: str,
+    dimension: str,
+    session: Session = Depends(get_session),
+) -> JSONResponse:
+    """
+    Add a dimension to a column on a node
+    """
+    node = get_node_by_name(session=session, name=name)
+    dimension_node = get_node_by_name(
+        session=session,
+        name=dimension,
+        node_type=NodeType.DIMENSION,
+    )
+
+    target_column = None
+    for node_column in node.current.columns:
+        if node_column.name == column:
+            target_column = node_column
+            node_column.dimension = dimension_node
+            node_column.dimension_id = dimension_node.id
+    if not target_column:
+        raise DJException(
+            message=f"A column with name `{column}` on node `{name}` does not exist.",
+            http_status_code=404,
+        )
+
+    session.add(node)
+    session.commit()
+    session.refresh(node)
+    return JSONResponse(
+        status_code=200,
+        content={
+            "message": (
+                f"Dimension node {dimension} has been successfully "
+                f"linked to column {column} on node {name}"
+            ),
+        },
+    )
+
+
+@router.post("/nodes/{name}/table/")
+def add_table_to_node(
+    name: str, data: CreateTable, *, session: Session = Depends(get_session)
```

Comment:
Will the table always be a new table? Is it possible that someone would add a table that already exists and thus we should check to see if we already have records of the table and then only add it if needed?

Reply:
Great catch, I'll add a check for an existing table on the given node.